## Loading the Standard libraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [10]:
data = pd.read_csv('zomato.csv', encoding = 'latin1')
data.head()

Restaurant ID         Restaurant Name  Country Code              City  \
0        6317637        Le Petit Souffle           162       Makati City   
1        6304287        Izakaya Kikufuji           162       Makati City   
2        6300002  Heat - Edsa Shangri-La           162  Mandaluyong City   
3        6318506                    Ooma           162  Mandaluyong City   
4        6314302             Sambo Kojin           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   
3  Third Floor, Mega Fashion Hall, SM Megamall, O...   
4  Third Floor, Mega Atrium, SM Megamall, Ortigas...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   
3      SM Megamall, Ortigas, Mandaluyong City   
4      SM Megamall, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   
3  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.056475  14.585318   
4  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.057508  14.584450   

                           Cuisines  ...          Currency Has Table booking  \
0        French, Japanese, Desserts  ...  Botswana Pula(P)               Yes   
1                          Japanese  ...  Botswana Pula(P)               Yes   
2  Seafood, Asian, Filipino, Indian  ...  Botswana Pula(P)               Yes   
3                   Japanese, Sushi  ...  Botswana Pula(P)                No   
4                  Japanese, Korean  ...  Botswana Pula(P)               Yes   

  Has Online delivery Is delivering now Switch to order menu Price range  \
0                  No                No                   No           3   
1                  No                No                   No           3   
2                  No                No                   No           4   
3                  No                No                   No           4   
4                  No                No                   No           4   

   Aggregate rating  Rating color Rating text Votes  
0               4.8    Dark Green   Excellent   314  
1               4.5    Dark Green   Excellent   591  
2               4.4         Green   Very Good   270  
3               4.9    Dark Green   Excellent   365  
4               4.8    Dark Green   Excellent   229  

[5 rows x 21 columns]

## checking the columns available in the data

In [11]:
data.columns

Index(['Restaurant ID', 'Restaurant Name', 'Country Code', 'City', 'Address',
       'Locality', 'Locality Verbose', 'Longitude', 'Latitude', 'Cuisines',
       'Average Cost for two', 'Currency', 'Has Table booking',
       'Has Online delivery', 'Is delivering now', 'Switch to order menu',
       'Price range', 'Aggregate rating', 'Rating color', 'Rating text',
       'Votes'],
      dtype='object')

## Total rows and total columns in the data

In [12]:
data.shape

(9551, 21)

- Observation :
1. There are 9551 rows and 21 columns in the data

### Delete unnecessary columns from the data

In [13]:
cols_to_del = ['Country Code', 'Address', 'Locality', 'Locality Verbose', 'Longitude', 'Latitude',
              'Switch to order menu', 'Rating color']
cols_to_del

['Country Code',
 'Address',
 'Locality',
 'Locality Verbose',
 'Longitude',
 'Latitude',
 'Switch to order menu',
 'Rating color']

In [15]:
data = data.drop(cols_to_del, axis = 1)
data.head()

Restaurant ID         Restaurant Name              City  \
0        6317637        Le Petit Souffle       Makati City   
1        6304287        Izakaya Kikufuji       Makati City   
2        6300002  Heat - Edsa Shangri-La  Mandaluyong City   
3        6318506                    Ooma  Mandaluyong City   
4        6314302             Sambo Kojin  Mandaluyong City   

                           Cuisines  Average Cost for two          Currency  \
0        French, Japanese, Desserts                  1100  Botswana Pula(P)   
1                          Japanese                  1200  Botswana Pula(P)   
2  Seafood, Asian, Filipino, Indian                  4000  Botswana Pula(P)   
3                   Japanese, Sushi                  1500  Botswana Pula(P)   
4                  Japanese, Korean                  1500  Botswana Pula(P)   

  Has Table booking Has Online delivery Is delivering now  Price range  \
0               Yes                  No                No            3   
1               Yes                  No                No            3   
2               Yes                  No                No            4   
3                No                  No                No            4   
4               Yes                  No                No            4   

   Aggregate rating Rating text  Votes  
0               4.8   Excellent    314  
1               4.5   Excellent    591  
2               4.4   Very Good    270  
3               4.9   Excellent    365  
4               4.8   Excellent    229

In [16]:
data.columns

Index(['Restaurant ID', 'Restaurant Name', 'City', 'Cuisines',
       'Average Cost for two', 'Currency', 'Has Table booking',
       'Has Online delivery', 'Is delivering now', 'Price range',
       'Aggregate rating', 'Rating text', 'Votes'],
      dtype='object')

### Dropping Duplicate rows from the data

In [18]:
data = data.drop_duplicates()
data.head()

Restaurant ID         Restaurant Name              City  \
0        6317637        Le Petit Souffle       Makati City   
1        6304287        Izakaya Kikufuji       Makati City   
2        6300002  Heat - Edsa Shangri-La  Mandaluyong City   
3        6318506                    Ooma  Mandaluyong City   
4        6314302             Sambo Kojin  Mandaluyong City   

                           Cuisines  Average Cost for two          Currency  \
0        French, Japanese, Desserts                  1100  Botswana Pula(P)   
1                          Japanese                  1200  Botswana Pula(P)   
2  Seafood, Asian, Filipino, Indian                  4000  Botswana Pula(P)   
3                   Japanese, Sushi                  1500  Botswana Pula(P)   
4                  Japanese, Korean                  1500  Botswana Pula(P)   

  Has Table booking Has Online delivery Is delivering now  Price range  \
0               Yes                  No                No            3   
1               Yes                  No                No            3   
2               Yes                  No                No            4   
3                No                  No                No            4   
4               Yes                  No                No            4   

   Aggregate rating Rating text  Votes  
0               4.8   Excellent    314  
1               4.5   Excellent    591  
2               4.4   Very Good    270  
3               4.9   Excellent    365  
4               4.8   Excellent    229

In [19]:
data.shape

(9551, 13)

### Proportion of Excellent and Very Good rated restaurants

In [17]:
data['Rating text'].value_counts(normalize = True) * 100

Average      39.126793
Not rated    22.489792
Good         21.987226
Very Good    11.297246
Excellent     3.151502
Poor          1.947440
Name: Rating text, dtype: float64

### Observations:
1. Lot of Resaurants have been rated as average(39.12% )
2. Only 3.15% restaurants are Excellent rated 
3. 11.29 % of restaurants rated as Very Good

### Total count of restaurants having aggregate rating greater than or equal to 4

In [22]:
data[data['Aggregate rating'] >= 4]['Restaurant Name'].count()

1380

In [23]:
1380 / 9551 * 100

14.448748822112867

#### Observations :
- Out of 9551 restaurants only 1380 have aggregate ratings >= 4 (14.44%)

### Analyzing City variable

In [24]:
data['City'].value_counts()

New Delhi           5473
Gurgaon             1118
Noida               1080
Faridabad            251
Ghaziabad             25
                    ... 
Panchkula              1
Mc Millan              1
Mayfield               1
Macedon                1
Vineland Station       1
Name: City, Length: 141, dtype: int64

### Proportion of Text ratings in New Delhi

In [29]:
data[data['City'] == 'New Delhi']['Rating text'].value_counts(normalize = True) * 100

Average      45.587429
Not rated    26.036908
Good         20.610269
Very Good     5.481454
Poor          1.772337
Excellent     0.511602
Name: Rating text, dtype: float64

### Observations :

- 45.58% of restaurants in New Delhi are average rated

### Count of restaurants in new delhi with aggregate rating >= 4

In [31]:
data[(data['City'] == 'New Delhi') & (data['Aggregate rating'] >= 4)]['Restaurant Name'].count()

328

#### Only 328 out of 5473 restaurants are having aggregate rating >= 4

In [32]:
328 / 5473 * 100

5.993056824410744

## Consider Cuisines variable

In [33]:
data.head()

Restaurant ID         Restaurant Name              City  \
0        6317637        Le Petit Souffle       Makati City   
1        6304287        Izakaya Kikufuji       Makati City   
2        6300002  Heat - Edsa Shangri-La  Mandaluyong City   
3        6318506                    Ooma  Mandaluyong City   
4        6314302             Sambo Kojin  Mandaluyong City   

                           Cuisines  Average Cost for two          Currency  \
0        French, Japanese, Desserts                  1100  Botswana Pula(P)   
1                          Japanese                  1200  Botswana Pula(P)   
2  Seafood, Asian, Filipino, Indian                  4000  Botswana Pula(P)   
3                   Japanese, Sushi                  1500  Botswana Pula(P)   
4                  Japanese, Korean                  1500  Botswana Pula(P)   

  Has Table booking Has Online delivery Is delivering now  Price range  \
0               Yes                  No                No            3   
1               Yes                  No                No            3   
2               Yes                  No                No            4   
3                No                  No                No            4   
4               Yes                  No                No            4   

   Aggregate rating Rating text  Votes  
0               4.8   Excellent    314  
1               4.5   Excellent    591  
2               4.4   Very Good    270  
3               4.9   Excellent    365  
4               4.8   Excellent    229

In [34]:
data['Cuisines']

0             French, Japanese, Desserts
1                               Japanese
2       Seafood, Asian, Filipino, Indian
3                        Japanese, Sushi
4                       Japanese, Korean
                      ...               
9546                             Turkish
9547     World Cuisine, Patisserie, Cafe
9548              Italian, World Cuisine
9549                     Restaurant Cafe
9550                                Cafe
Name: Cuisines, Length: 9551, dtype: object

In [35]:
data[['Restaurant Name', 'Cuisines']]

Restaurant Name                          Cuisines
0             Le Petit Souffle        French, Japanese, Desserts
1             Izakaya Kikufuji                          Japanese
2       Heat - Edsa Shangri-La  Seafood, Asian, Filipino, Indian
3                         Ooma                   Japanese, Sushi
4                  Sambo Kojin                  Japanese, Korean
...                        ...                               ...
9546              NamlÛ± Gurme                           Turkish
9547             Ceviz AÛôacÛ±   World Cuisine, Patisserie, Cafe
9548                     Huqqa            Italian, World Cuisine
9549              Aôôk Kahve                   Restaurant Cafe
9550  Walter's Coffee Roastery                              Cafe

[9551 rows x 2 columns]

In [36]:
data.head()

Restaurant ID         Restaurant Name              City  \
0        6317637        Le Petit Souffle       Makati City   
1        6304287        Izakaya Kikufuji       Makati City   
2        6300002  Heat - Edsa Shangri-La  Mandaluyong City   
3        6318506                    Ooma  Mandaluyong City   
4        6314302             Sambo Kojin  Mandaluyong City   

                           Cuisines  Average Cost for two          Currency  \
0        French, Japanese, Desserts                  1100  Botswana Pula(P)   
1                          Japanese                  1200  Botswana Pula(P)   
2  Seafood, Asian, Filipino, Indian                  4000  Botswana Pula(P)   
3                   Japanese, Sushi                  1500  Botswana Pula(P)   
4                  Japanese, Korean                  1500  Botswana Pula(P)   

  Has Table booking Has Online delivery Is delivering now  Price range  \
0               Yes                  No                No            3   
1               Yes                  No                No            3   
2               Yes                  No                No            4   
3                No                  No                No            4   
4               Yes                  No                No            4   

   Aggregate rating Rating text  Votes  
0               4.8   Excellent    314  
1               4.5   Excellent    591  
2               4.4   Very Good    270  
3               4.9   Excellent    365  
4               4.8   Excellent    229

In [37]:
data['Currency'].unique()

array(['Botswana Pula(P)', 'Brazilian Real(R$)', 'Dollar($)',
       'Emirati Diram(AED)', 'Indian Rupees(Rs.)',
       'Indonesian Rupiah(IDR)', 'NewZealand($)', 'Pounds(\x8c£)',
       'Qatari Rial(QR)', 'Rand(R)', 'Sri Lankan Rupee(LKR)',
       'Turkish Lira(TL)'], dtype=object)

In [39]:
zomato_india = data[data['Currency'] == 'Indian Rupees(Rs.)']
zomato_india.head()

Restaurant ID                        Restaurant Name  City  \
624        3400025                             Jahanpanah  Agra   
625        3400341                    Rangrezz Restaurant  Agra   
626        3400005                Time2Eat - Mama Chicken  Agra   
627        3400021  Chokho Jeeman Marwari Jain Bhojanalya  Agra   
628        3400017                         Pinch Of Spice  Agra   

                           Cuisines  Average Cost for two            Currency  \
624           North Indian, Mughlai                   850  Indian Rupees(Rs.)   
625           North Indian, Mughlai                   700  Indian Rupees(Rs.)   
626                    North Indian                   500  Indian Rupees(Rs.)   
627                      Rajasthani                   400  Indian Rupees(Rs.)   
628  North Indian, Chinese, Mughlai                  1000  Indian Rupees(Rs.)   

    Has Table booking Has Online delivery Is delivering now  Price range  \
624                No                  No                No            3   
625                No                  No                No            2   
626                No                  No                No            2   
627                No                  No                No            2   
628                No                  No                No            3   

     Aggregate rating Rating text  Votes  
624               3.9        Good    140  
625               3.5        Good     71  
626               3.6        Good     94  
627               4.0   Very Good     87  
628               4.2   Very Good    177

In [40]:
zomato_india.shape

(8652, 13)

### Total count of cuisines in the indian restaurants

In [42]:
zomato_india['Cuisines'].count()

8652

In [43]:
zomato_india['Cuisines'].unique()

array(['North Indian, Mughlai', 'North Indian', 'Rajasthani', ...,
       'Continental, Seafood, Chinese, North Indian, Biryani',
       'Burger, Pizza, Biryani',
       'American, North Indian, Thai, Continental'], dtype=object)

In [44]:
zomato_india['Cuisines'].nunique()

1392

In [ ]:
# zomato_india['Cuisines'] = zomato_india['Cuisines'].apply(lambda x : x.str.split(','))

In [45]:
zomato_india['Cuisines'] = zomato_india['Cuisines'].str.split(',')

In [46]:
zomato_india.head()

Restaurant ID                        Restaurant Name  City  \
624        3400025                             Jahanpanah  Agra   
625        3400341                    Rangrezz Restaurant  Agra   
626        3400005                Time2Eat - Mama Chicken  Agra   
627        3400021  Chokho Jeeman Marwari Jain Bhojanalya  Agra   
628        3400017                         Pinch Of Spice  Agra   

                               Cuisines  Average Cost for two  \
624            [North Indian,  Mughlai]                   850   
625            [North Indian,  Mughlai]                   700   
626                      [North Indian]                   500   
627                        [Rajasthani]                   400   
628  [North Indian,  Chinese,  Mughlai]                  1000   

               Currency Has Table booking Has Online delivery  \
624  Indian Rupees(Rs.)                No                  No   
625  Indian Rupees(Rs.)                No                  No   
626  Indian Rupees(Rs.)                No                  No   
627  Indian Rupees(Rs.)                No                  No   
628  Indian Rupees(Rs.)                No                  No   

    Is delivering now  Price range  Aggregate rating Rating text  Votes  
624                No            3               3.9        Good    140  
625                No            2               3.5        Good     71  
626                No            2               3.6        Good     94  
627                No            2               4.0   Very Good     87  
628                No            3               4.2   Very Good    177

## Split the Cuisines columns into seperate cuisines and check all unique cuisines available in india

## How many restaurants have online delivery servies?

In [52]:
zomato_india.groupby('Has Online delivery')['Restaurant Name'].agg('count') / len(zomato_india) * 100

Has Online delivery
No     71.994914
Yes    28.005086
Name: Restaurant Name, dtype: float64

In [53]:
zomato_india['City'].unique()

array(['Agra', 'Ahmedabad', 'Allahabad', 'Amritsar', 'Aurangabad',
       'Bangalore', 'Bhopal', 'Bhubaneshwar', 'Chandigarh', 'Chennai',
       'Coimbatore', 'Dehradun', 'Faridabad', 'Ghaziabad', 'Goa',
       'Gurgaon', 'Guwahati', 'Hyderabad', 'Indore', 'Jaipur', 'Kanpur',
       'Kochi', 'Kolkata', 'Lucknow', 'Ludhiana', 'Mangalore', 'Mohali',
       'Mumbai', 'Mysore', 'Nagpur', 'Nashik', 'New Delhi', 'Noida',
       'Panchkula', 'Patna', 'Puducherry', 'Pune', 'Ranchi',
       'Secunderabad', 'Surat', 'Vadodara', 'Varanasi', 'Vizag'],
      dtype=object)

### Average cost for two per city

In [57]:
zomato_india.groupby('City')[[ 'Average Cost for two']].agg(['min', 'max'])

Average Cost for two      
                              min   max
City                                   
Agra                            0  3600
Ahmedabad                     300  1800
Allahabad                       0  1000
Amritsar                      100  1200
Aurangabad                    300  1800
Bangalore                     400  2400
Bhopal                        200  1800
Bhubaneshwar                  300  1500
Chandigarh                    450  2200
Chennai                       350  2000
Coimbatore                    300  1500
Dehradun                        0  1600
Faridabad                     100  1900
Ghaziabad                     100  2500
Goa                           600  2000
Gurgaon                       100  5000
Guwahati                      300  1650
Hyderabad                     500  2000
Indore                        250  2000
Jaipur                        650  2500
Kanpur                          0  3000
Kochi                         350  1500
Kolkata                       800  1900
Lucknow                       200  2000
Ludhiana                      400  2000
Mangalore                     200  1600
Mohali                        550   550
Mumbai                        500  1500
Mysore                        290  1700
Nagpur                        250  1600
Nashik                        100  1500
New Delhi                      50  8000
Noida                          50  3500
Panchkula                    2000  2000
Patna                         300  1600
Puducherry                    150  1500
Pune                          800  2100
Ranchi                        200  2000
Secunderabad                  800   850
Surat                         250  1500
Vadodara                      300  1500
Varanasi                        0  1100
Vizag                         300  1700

In [58]:
zomato_india.head()

Restaurant ID                        Restaurant Name  City  \
624        3400025                             Jahanpanah  Agra   
625        3400341                    Rangrezz Restaurant  Agra   
626        3400005                Time2Eat - Mama Chicken  Agra   
627        3400021  Chokho Jeeman Marwari Jain Bhojanalya  Agra   
628        3400017                         Pinch Of Spice  Agra   

                               Cuisines  Average Cost for two  \
624            [North Indian,  Mughlai]                   850   
625            [North Indian,  Mughlai]                   700   
626                      [North Indian]                   500   
627                        [Rajasthani]                   400   
628  [North Indian,  Chinese,  Mughlai]                  1000   

               Currency Has Table booking Has Online delivery  \
624  Indian Rupees(Rs.)                No                  No   
625  Indian Rupees(Rs.)                No                  No   
626  Indian Rupees(Rs.)                No                  No   
627  Indian Rupees(Rs.)                No                  No   
628  Indian Rupees(Rs.)                No                  No   

    Is delivering now  Price range  Aggregate rating Rating text  Votes  
624                No            3               3.9        Good    140  
625                No            2               3.5        Good     71  
626                No            2               3.6        Good     94  
627                No            2               4.0   Very Good     87  
628                No            3               4.2   Very Good    177

## Check whether any restaurant having aggregate rating = 5

In [61]:
zomato_india[zomato_india['Aggregate rating'] == 5.0]

Empty DataFrame
Columns: [Restaurant ID, Restaurant Name, City, Cuisines, Average Cost for two, Currency, Has Table booking, Has Online delivery, Is delivering now, Price range, Aggregate rating, Rating text, Votes]
Index: []

- Observation:
1. No restaurant has an average rating of 5.0

## check how many restaurants have both online delivery and table booking 

### Create a Price range bracket using the Average cost for two columns
a. 0 - 1000 = very cheap
b. 1000 - 2000 = cheap
c. 2000 - 3000 = affordable
d. 3000 - 4000 = Costly
e. 5000 & above = Expensive

### check the restaurant name where the Average Cost for Two people is 8000

In [64]:
zomato_india[zomato_india['Average Cost for two'] == 8000]

Restaurant ID                    Restaurant Name       City    Cuisines  \
7543           2701  Orient Express - Taj Palace Hotel  New Delhi  [European]   

      Average Cost for two            Currency Has Table booking  \
7543                  8000  Indian Rupees(Rs.)               Yes   

     Has Online delivery Is delivering now  Price range  Aggregate rating  \
7543                  No                No            4               4.0   

     Rating text  Votes  
7543   Very Good    145